src/main/java/uk/gov/pay/api/filter/ratelimit/RedisRateLimiter.java

```diff
@@ -14,37 +15,38 @@
 public class RedisRateLimiter {
     private static final Logger LOGGER = LoggerFactory.getLogger(RedisRateLimiter.class);
 
-    private final int noOfReq;
-    private final int noOfReqForPost;
+    private RateLimitManager rateLimitManager;
     private final int perMillis;
     private JedisPool jedisPool;
 
     @Inject
-    public RedisRateLimiter(int noOfReq, int noOfReqForPost, int perMillis, JedisPool jedisPool) {
-        this.noOfReq = noOfReq;
-        this.noOfReqForPost = noOfReqForPost;
+    public RedisRateLimiter(RateLimitManager rateLimitManager, int perMillis, JedisPool jedisPool) {
+        this.rateLimitManager = rateLimitManager;
         this.perMillis = perMillis;
         this.jedisPool = jedisPool;
     }
 
     /**
      * @throws RateLimitException
      */
-    synchronized void checkRateOf(String accountId, String key, String method)
+    synchronized void checkRateOf(String accountId, RateLimiterKey key)
             throws RedisException, RateLimitException {
 
         Long count;
 
         try {
-            count = updateAllowance(key);
+            count = updateAllowance(key.getKey());
         } catch (Exception e) {
             // Exception possible if redis is unavailable or perMillis is too high        
             throw new RedisException();
         }
 
-        if (count != null && count > getNoOfReqForMethod(method)) {
-            LOGGER.info(String.format("RedisRateLimiter - Rate limit exceeded for account [%s] and method [%s] - count: %d, rate allowed: %d", accountId, method, count, getNoOfReqForMethod(method)));
-            throw new RateLimitException();
+        if (count != null) {
+            int allowedNumberOfRequests = rateLimitManager.getAllowedNumberOfRequests(key, accountId);
+            if (count > allowedNumberOfRequests) {
+                LOGGER.info(String.format("RedisRateLimiter - Rate limit exceeded for account [%s] and type [%s] - count: %d, rate allowed: %d", accountId, key.getKeyType(), count, allowedNumberOfRequests));
```

Comment:
Would be useful to log path too (capture/payments..). Could come in handy when we move to structured logging

Reply:
I would prefer `method / http method` over `type`

Reply:
it's not always `POST` sometimes it's `POST-create_payment`, so maybe method is fine.

Reply:
it's not always `POST` sometimes it's `POST-create_payment`, so maybe method is fine.